op-challenger/game/fault/preimages/large.go

```diff
@@ -31,24 +52,72 @@ func NewLargePreimageUploader(logger log.Logger, txMgr txmgr.TxManager, contract
 	return &LargePreimageUploader{logger, txMgr, contract}
 }
 
+// GetKeccakLeaf returns the keccak leaf for the preimage data at the specified offset.
+func GetKeccakLeaf(data *types.PreimageOracleData, offset uint32) []byte {
+	offset = offset * matrix.LeafSize
+	end := offset + matrix.LeafSize
+	if offset >= uint32(len(data.OracleData)) {
+		return make([]byte, matrix.LeafSize)
+	}
+	if end > uint32(len(data.OracleData)) {
+		padded := make([]byte, matrix.LeafSize)
+		data := data.OracleData[offset:]
+		copy(padded[matrix.LeafSize-len(data):], data)
+		return padded
+	}
+	return data.OracleData[offset:end]
+}
```

Comment:
This seems to be duplicating the splitting logic in `AbsorbNextLeaf`. Seems like it would be better for `AbsorbNextLeaf` to simply return the data it read?